tools/dispatch/__init__.py

```diff
@@ -0,0 +1 @@
+../dispatch.sh
```

Comment:
Does it make any sense to have user overrides for things like `__init__.py` or `extract_platinfo.py`?

Also, IIUC, adding new tools will require adding dispatch entries - arguably that can be scripted, but it just feels like unnecessary trouble.

What if we go for a e.g. `LISA_USER_BIN` env var, and then we can have something like:
```bash
_LISA_PATH="$LISA_HOME/shell/:$LISA_HOME/tools/$LISA_HOST_ABI:$LISA_HOME/tools/:$ANDROID_HOME/tools/:$ANDROID_HOME/platform-tools/"

if [[ -n $LISER_USER_BIN ]]; then
        # User bins take priority
        export PATH=$PATH:$_LISA_PATH
else
        # LISA bins take priority
        export PATH=$_LISA_PATH:$PATH
fi
```

That way, first-time users get the right setup, and existing users can just add a single env var to their .bashrc and keep on using their hodgepodge of binaries.

Reply:
I haven't added any symlinks for programs we ship, so `extract_platinfo.py` is not part of them, for the reason you state.

> Also, IIUC, adding new tools will require adding dispatch entries - arguably that can be scripted, but it just feels like unnecessary trouble.

They will need to be symlinked in the dispatch folder indeed. One good side is that we know exactly the list of things that we add in the PATH ...

> What if we go for a e.g. LISA_USER_BIN env var, and then we can have something like:

That's what I proposed at some point, however that has also its flaws: people might want the adb coming from the android SDK, and the trace-cmd of their distro at the same time. Settings like that conflict with each other and can only be solved by a per-binary decision, which is what this commit brings in a compact way.

Reply:
following offline discussion, we can adopt that solution now and if we need more flexibility in the future, we can always add that dispatch implementation then.